packages/components/src/components/form/IconButton/IconButton.tsx

```diff
@@ -13,5 +13,12 @@ export const IconButton: FC<IconButtonProps> = ({ isRound, icon, ...props }) =>
 
   const Icon = React.cloneElement(icon, { size: props.size });
 
-  return <ChakraIconButton spinner={<Spinner {...iconButtonLoading} />} icon={Icon} {...iconButtonStyles} {...props} />;
+  return (
+    <ChakraIconButton
+      spinner={<Spinner {...iconButtonLoading} variant={props.variant as SpinnerVariant} />}
```

Comment:
this typecast is wrong, there is no 1:1 match between the variants of the Spinner and the Button